src/_css/main.css

```diff
@@ -27,6 +37,45 @@
 
   /* Body padding offset. */
   --offset: 2em;
+
+  /* Light mode */
+  --html-color: var(--black);
+  --html-background-color: var(--darkgray);
+  --main-color: var(--black);
+  --main-background-color: var(--white);
+  --footer-background-color: var(--lightestgray);
+  --main-and-footer-link-color: var(--darkblue);
+  --avatar-img-filter: none;
+  --icon-no-support-filter: grayscale(100%);
+  --note-color: var(--main-color);
+  --note-background-color: var(--lightyellow);
+  --tag-color: var(--black);
+  --tag-background-color: var(--lightestgray);
+  --logo-hover-background-color: var(--lightestgray);
+}
+
+:root.dark {
```

Comment:
Is there a nice way to make this work for `@media (prefers-color-scheme: dark) {…}` as well? So that, if the JS fails to load for some reason, we still get the light/dark mode depending on OS preference?

Reply:
> Is there a nice way to make this work for `@media (prefers-color-scheme: dark) {…}` as well? So that, if the JS fails to load for some reason, we still get the light/dark mode depending on OS preference?

Unfortunately not, we'd have to roll back to https://github.com/v8/v8.dev/pull/254/commits/4ae061636329bd0184ddd64fc6b878d783dbfe57 with the multiple CSS files to make this completely independent from JS. With the class approach (one CSS file), the custom element listens for `prefers-color-scheme` changes and then [acts](https://github.com/v8/v8.dev/blob/e82bb820e649c847670cdac171ebf4c56e84806b/src/_js/main.mjs#L25-L29) upon them.

Reply:
I guess there _could_ be a solution in theory, where `<dark-mode-toggle>` would fetch the CSS file, inspect the contents to figure out what's wrapped in `@media (prefers-color-scheme: dark) { … }`, and then do its magic that way.

But I agree, the current approach seems fine.

Reply:
> I guess there could be a solution in theory, where `<dark-mode-toggle>` would fetch the CSS file

Wait, I thought the constraint was that its *"JS fails to load for some reason"*… :-)

Reply:
> > I guess there could be a solution in theory, where `<dark-mode-toggle>` would fetch the CSS file
> 
> Wait, I thought the constraint was that its *"JS fails to load for some reason"*… :-) 

Yes, that is the constraint. I'm probably not explaining my thoughts very well, apologies :P

With a pure CSS approach using `@media (prefers-color-scheme: dark) { … }`, **everything works even without JS**. This is the standard approach.

I'm saying that `<dark-mode-toggle>` could (theoretically) build on top of this, without the need for the earlier approach that included HTML + additional `<script>` changes, and without special hardcoded CSS classes (the current approach) that still rely on JS.

To be clear, I don't want to block this patch on this. I'm fine with the current approach. I'm just thinking out loud here :)

Reply:
The way to make this work with *and* without JavaScript would be as follows:

- **With JavaScript** (i.e., relying on `prefers-color-scheme` with the toggle to override it):

   Just do what we do in https://github.com/v8/v8.dev/pull/254/commits/e82bb820e649c847670cdac171ebf4c56e84806b. The reason why we use a hard-coded `dark` class (or in the previous multiple CSS files approach, override the `media` attribute values) is that we explicitly *fight* the browser; we override the system preference it communicates down to us.

- **Without JavaScript** (i.e., relying on `prefers-color-scheme` alone):

   Wrap the [CSS rules relevant for dark mode](https://github.com/v8/v8.dev/blob/e82bb820e649c847670cdac171ebf4c56e84806b/src/_css/main.css#L57-L79) as follows (we would want to automate this probably):

   ```html
   <noscript>
     <style>
       @media (prefers-color-scheme: dark) {
         /* Inline https://github.com/v8/v8.dev/blob/e82bb820e649c847670cdac171ebf4c56e84806b/src/_css/main.css#L57-L79 */
         /* This could of course also be outsourced to an external file */
       }
     </style>
   </noscript>
   ```